spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/MqttPahoMessageHandler.java

```diff
@@ -208,7 +204,7 @@ private void sendDeliveryComplete(IMqttDeliveryToken token) {
 	}
 
 	@Override
-	public void connectionLost(Throwable cause) {
+	public synchronized void connectionLost(Throwable cause) {
```

Comment:
OK. I think I got it: https://docs.oracle.com/javase/tutorial/essential/concurrency/syncmeth.html
The `synchronized` on method is the same like:
```
synchronized (this) {

}
```
as a first expression in the method.

Merging as is, but with the `@SuppressWarnings("deprecation")`, of course :smile: